From 27fa3e3f8a56c793ff317f1007ffc611514ee6fa Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Fri, 27 May 2022 14:07:39 +0800
Subject: [PATCH] pfe: sw: drop the unneeded codes to fix build errors

This patch is to fix the following build error:

  error: the comparison will always evaluate as 'true' for the address of 'macaddr' will never be NULL [-Werror=address]
     942 |         if (netif->cfg->macaddr && is_valid_ether_addr(netif->cfg->macaddr))
         |             ^~~~~

The reason of this issue is that, the gcc compiler of CD release is updated
to v12.1.0 version, and the above build error shows.

This patch is to fix it by dropping the unneeded codes.

Upstream-Status: Pending

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 sw/linux-pfeng/pfeng-netif.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sw/linux-pfeng/pfeng-netif.c b/sw/linux-pfeng/pfeng-netif.c
index 15d7a46..656abc8 100644
--- a/sw/linux-pfeng/pfeng-netif.c
+++ b/sw/linux-pfeng/pfeng-netif.c
@@ -939,7 +939,7 @@ static int pfeng_netif_logif_init_second_stage(struct pfeng_netif *netif)
 		goto err;
 
 	/* Set MAC address */
-	if (netif->cfg->macaddr && is_valid_ether_addr(netif->cfg->macaddr))
+	if (is_valid_ether_addr(netif->cfg->macaddr))
 		memcpy(&saddr.sa_data, netif->cfg->macaddr, sizeof(saddr.sa_data));
 	else
 		memset(&saddr.sa_data, 0, sizeof(saddr.sa_data));
-- 
2.25.1

